lib/nationbuilder/client.rb

```diff
@@ -92,6 +92,8 @@ def perform_request_with_retries(method, url, request_args)
       else
         exception_to_reraise = nil
         break
+      ensure
+        set_response(raw_response)
```

Comment:
Because of the assignment to temporary variables, I can see potential here for `raw_response` and `parsed_response` to be out of sync, e.g. if the `raw_response = ` worked, but the `parsed_response = ` raised an exception. I wonder if something like this might be safer?

```ruby
def perform_request_with_retries(method, url, request_args)
  (@retries + 1).times do |i|
    exception_to_reraise = nil
    
    begin
      set_response(@http_client.send(method, url, request_args))
    rescue NationBuilder::RateLimitedError => e
      exception_to_reraise = e
      Kernel.sleep(RETRY_DELAY * 2**i)
    rescue => e
      raise e
    end
  end

  raise exception_to_reraise if exception_to_reraise
  
  parse_response_body(response)
end
```

I haven't tested this, but it's a rough sketch of an approach that I believe would guarantee the parsed response returned is a function of the value of the thread local.